client/src/components/SearchBar/SearchBar.tsx

```diff
@@ -0,0 +1,80 @@
+import React, { useEffect, useState } from "react";
+import { OutlinedInput, InputLabel, InputAdornment } from "@material-ui/core";
+import LocationOnIcon from "@material-ui/icons/LocationOn";
+import LocationOffIcon from "@material-ui/icons/LocationOff";
+import { isDebug } from "src/config";
+
+function SearchBar({
+  iconCallback,
+  onChangeCallback,
+  onEnterCallback,
+}: {
+  iconCallback?: (status: boolean) => void;
+  onChangeCallback?: (text: string) => void;
+  onEnterCallback?: (text: string) => void;
+}) {
+  const [gpsStatus, setGpsStatus] = useState(false);
+
+  const iconClickWrapper = () => {
+    setGpsStatus(!gpsStatus);
+  };
+
+  const dynamicChangeWrapper = (
+    event: React.ChangeEvent<HTMLInputElement | HTMLTextAreaElement>
+  ) => {
+    if (onChangeCallback) {
+      onChangeCallback(event.target.value);
+    }
+  };
+
+  const keyPressWrapper = (event: React.KeyboardEvent<HTMLDivElement>) => {
+    if (event.key === "Enter") {
+      const ele = document.getElementById("stores-search-bar");
+      if (ele) ele.blur();
```

Comment:
should this be the blurWrapper?

Reply:
Nope, I'm attempting to trigger a blur event for the presently focused text input when user presses ENTER

Reply:
got it, perhaps the keyPressWrapper name can be checked to triggerBlurOnKeypress or something

Reply:
left comment to clarify intention, we might need this wrapper to do something else other than trigger blur too (since it triggers on all keypresses)